tasks/lib/auth.js

```diff
@@ -52,6 +52,9 @@ AuthUtil.handlers.oauth = function(grunt, service, envs, formatCreds, cb) {
   var cbRoute = route + '/callback';
 
   // Calculate the callbackURL for the request
+  if (process.env.OAUTH_SERVER_PORT == 0) {
+    process.env.OAUTH_SERVER_PORT = process.env.PORT;
```

Comment:
Could you try something on your C9 IDE?

Change your .env file to

```
OAUTH_SERVER_PORT=$PORT
```

I _think_ this should pull in the C9 port number and assign it to the `OAUTH_SERVER_PORT` var, so we don't need this extra bit of code.